lib/internal/modules/esm/loader.js

```diff
@@ -21,16 +21,28 @@ const {
 } = require('internal/modules/esm/utils');
 let defaultResolve, defaultLoad, defaultLoadSync, importMetaInitializer;
 
+/**
+ * Generate a resolve cache (to store the final result of a resolve-chain for a particular module).
```

Comment:
There are some inconsistencies when sometimes we use infinitive forms, sometimes we use third person. IMO the comment regarding what to use the resolve cache does not belong here, but rather on the `ResolveCache` class itself.
```suggestion
 * Lazy loads the module_map module and returns a new instance of ResolveCache.
```